pkg/reporter/insights-reporter.go

```diff
@@ -0,0 +1,246 @@
+	}
+
+	logrus.Infof("cluster %q not found... creating..", clusterName)
+
+	req, err = http.NewRequest("POST", clusterURL, nil)
+	if err != nil {
+		return nil, fmt.Errorf("building request for creating cluster: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+ir.auth.Token)
+	resp, err = http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("making request for creating cluster: %w", err)
+	}
+	defer resp.Body.Close()
+
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading response body: %w", err)
+	}
+
+	if !isSuccessful2XX(resp.StatusCode) {
+		return nil, fmt.Errorf("creating cluster, expected 200 OK received %s: %v", resp.Status, string(body))
+	}
+
+	var c insightsCluster
+	err = json.Unmarshal(body, &c)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshaling response body: %w", err)
+	}
+
+	logrus.Infof("cluster %q created...", clusterName)
+	return &c, nil
+}
+
+func (ir insightsReporter) sendReport(cluster *insightsCluster, reportType, reportVersion string, payload []byte) (*insightsReportJob, error) {
+	uploadReportURL := fmt.Sprintf("%s/v0/organizations/%s/clusters/%s/data/%s", ir.insightsHost, ir.auth.Organization, cluster.Name, reportType)
+	req, err := http.NewRequest("POST", uploadReportURL, bytes.NewBuffer(payload))
+	if err != nil {
+		return nil, fmt.Errorf("building request for output: %w", err)
+	}
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+cluster.AuthToken)
+	req.Header.Set("X-Fairwinds-Report-Version", reportVersion)
+	req.Header.Set("X-Fairwinds-Report-Priority", "4") // should have higher priority than the default 5
+
+	// TODO: Vitor - ?!
+	req.Header.Set("X-Fairwinds-Agent-Version", "")
+	req.Header.Set("X-Fairwinds-Agent-Chart-Version", "")
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return nil, fmt.Errorf("making request for output: %w", err)
+	}
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("reading response body: %w", err)
+	}
+	if !isSuccessful2XX(resp.StatusCode) {
+		return nil, fmt.Errorf("sending %s report, expected 200 OK received %s: %v", reportType, resp.Status, string(body))
+	}
+
+	var rj insightsReportJob
+	err = json.Unmarshal(body, &rj)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshaling response body: %w", err)
+	}
+	logrus.Debugf("%s report sent to fairwinds insights", reportType)
+
+	return &rj, nil
+}
+
+func isSuccessful2XX(statusCode int) bool {
+	return statusCode >= 200 && statusCode < 300
+}
+
+func (ir insightsReporter) getReportJob(clusterName string, reportJobID int) (*insightsReportJob, error) {
+	reportJobsURL := fmt.Sprintf("%s/v0/organizations/%s/clusters/%s/report-jobs/%d", ir.insightsHost, ir.auth.Organization, clusterName, reportJobID)
```

Comment:
might be worth making an "InsightsClient" pkg that has `Get` and `Post` operations to abstract out some of this logic

Reply:
done